src/frontend/app.ts

```diff
@@ -172,7 +173,15 @@ export default (octokitApp: App): Express => {
 	app.get("/jira/atlassian-connect.json", getJiraConnect);
 
 	// Maintenance mode view
-	app.use((req, res, next) => (isMaintenanceMode() ? getMaintenance(req, res) : next()));
+	app.use(async (req, res, next) => {
+		if (!req.session.jiraHost) {
+			// we only activate the maintenance mode for requests that are bound to a jira host
+			next();
+		} else {
+			await booleanFlag(BooleanFlags.MAINTENANCE_MODE, false, req.session.jiraHost) ? getMaintenance(req, res) : next();
```

Comment:
are we just the await is waiting for `booleanFlag` here? or is just waiting for the ternary operator output?

Reply:
I split it up now to make sure that we're only awaiting the `booleanFlag`.